agent/rpc/middleware/interceptors.go

```diff
@@ -49,7 +50,8 @@ func NewRequestRecorder(logger hclog.Logger, isLeader func() bool, localDC strin
 }
 
 func (r *RequestRecorder) Record(requestName string, rpcType string, start time.Time, request interface{}, respErrored bool) {
-	elapsed := time.Since(start).Milliseconds()
+	elapsed := time.Since(start).Microseconds()
+	elapsedMs := float64(elapsed) / math.Pow(10, 3)
```

Comment:
I think `float64` was originally used because of `math.Pow` but isn't necessary if we simply use 1000.

The following code should keep its precision
```suggestion
	elapsedMs := float32(elapsed) / 1000
```

Reply:
You are right. Thanks